### PR TITLE
[fix][broker] fix log error in exceptionCaught

### PR DESCRIPTION
**Pull request by TakaHiR07**
### Motivation

Fix log error in exceptionCaught

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by nodece — APPROVED** [review]

**Review by tisonkun — CHANGES_REQUESTED** [review]
I don't think it's reasonable. The extra last parameter for `log.warn` and `log.error` is for logging errors. See also `private void recordEvent(Level level, Marker marker, String msg, Object[] args, Throwable throwable) {`.

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public synchronized void channelInactive(ChannelHandlerContext ctx) throws Excep
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
         super.exceptionCaught(ctx, cause);
-        LOG.warn("[{}] Got exception {} : Message: {} State: {}", remoteAddress, cause.getClass().getSimpleName(),
+        LOG.warn("[{}] Got exception {} : Message: {} State: {} Cause: {}", remoteAddress, cause.getClass().getSimpleName(),
```
Comment (tisonkun) [inline pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java:212]:
```suggestion
        if (ClientCnx.isKnownException(cause)) {
            LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                    remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state, cause);
        } else {
            LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                    remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state);
        }
```

To stop the IDEA complaining, you can make this change.

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java [hunk 2]
```diff
@@ -209,7 +209,7 @@ public synchronized void channelInactive(ChannelHandlerContext ctx) throws Excep
...
                 cause.getMessage(), state,
                 ClientCnx.isKnownException(cause) ? null : cause);
```
Comment (tisonkun) [inline pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java:214]:
```suggestion
```suggestion
        if (ClientCnx.isKnownException(cause)) {
            LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                    remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state);
        } else {
            LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                    remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state, cause);
        }
```

To stop the IDEA complaining, you can make this change.
```

Comment (tisonkun) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java:214]: Or:

```java
        LOG.warn("[{}] Got exception {} : Message: {} State: {}",
                remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state,
                ClientCnx.filterKnownException(cause));

// in ClientCnx.java

    public static Throwable filterKnownException(Throwable t) {
        return isKnownException(t) ? null : t;
    }
```

It's somehow IDEA shortcoming.

Comment (TakaHiR07) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java:214]: > Or:
> 
> ```java
>         LOG.warn("[{}] Got exception {} : Message: {} State: {}",
>                 remoteAddress, cause.getClass().getSimpleName(), cause.getMessage(), state,
>                 ClientCnx.filterKnownException(cause));
> 
> // in ClientCnx.java
> 
>     public static Throwable filterKnownException(Throwable t) {
>         return isKnownException(t) ? null : t;
>     }
> ```
> 
> It's somehow IDEA shortcoming.

@tisonkun You are right, I am misled by idea complain.

**Pull request closed without merging**